frontend/lib/resolution/scope-queries.cpp

```diff
@@ -3031,9 +3113,11 @@ GatherMentionedModules::lookupAndGather(const Scope* scope,
   // if we aren't in a method, use a faster strategy to find the module
   if (!inMethod) {
     const Scope* resultScope = nullptr;
-    for (const auto& bids : v) {
-      for (const auto& id : bids) {
-        if (resultScope == nullptr && parsing::idIsModule(context, id)) {
+    auto end = v.end();
+    for (auto cur = v.begin(); cur != end; ++cur) {
```

Comment:
I see; this is what creates the need for a custom iterator. I think this can be cleaned up using an "adapter", but probably not in this PR. I'd rather have two ways to iterate using the "nice" `for` loop.

Reply:
In some ways, it'd be more natural for `for (const auto& x : myMatchingIdsWithName)` to yield `IdAndFlags`. However needing `ID` is common. I will add a TODO comment about this to `MatchingIdsWithName`.